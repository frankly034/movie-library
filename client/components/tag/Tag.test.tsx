import { fireEvent, render } from "@testing-library/react";

import Tag from ".";
describe("Component - Title", () => {
  let handleClick: () => void;

  beforeEach(() => {
    handleClick = jest.fn();
  });

  it("should render component", () => {
    const { container } = render(<Tag onClick={handleClick}>Action</Tag>);
    expect(container).toMatchSnapshot();
  });

  it("should render button component", () => {
    const { getByRole } = render(<Tag onClick={handleClick}>Adventure</Tag>);
    expect(getByRole("button")).toBeTruthy;
  });

  it("should fire the onClick function when clicked", () => {
    const { getByRole } = render(
      <Tag onClick={handleClick}>{"Click Me!"}</Tag>
    );

    fireEvent.click(getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
