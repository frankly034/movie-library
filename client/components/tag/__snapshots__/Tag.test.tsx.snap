// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Title should render component 1`] = `
<div>
  <button
    class="sc-bcPKhP TrbCn"
  >
    Action
  </button>
</div>
`;
